cloudapi/logs.go

```diff
@@ -176,3 +228,69 @@ func (c *Config) StreamLogsToLogger(
 		}
 	}
 }
+
+// timstampTrack is a safe-concurrent tracker
+// of the latest/most recent seen timestamp value.
+type timestampTrack struct {
+	// ts is timestamp in unix nano format
+	ts int64
+}
+
+// TimeOrNow returns as Time the latest tracked value plus 1ms
+// or Now as the default value.
+func (tst *timestampTrack) TimeOrNow() (t time.Time) {
+	t = time.Now()
+	if ts := atomic.LoadInt64(&tst.ts); ts > 0 {
+		// add 1ms for avoid possible repetition
+		t = time.Unix(0, ts).Add(1 * time.Millisecond)
+	}
```

Comment:
nit: I feel like this will be better of inline in the place it's used.

Currently, the whole timestampTrack is a 1 int64 that we atomically change under the hood and here you have some business logic on top. Given that it is literally set in 1 place and read in 1 place I would prefer if the business logic is more inlined and it's just that int64 that we atomically read and write, instead of it being something that I now need to go read that `TImeOrNow` (which I guess should be `ValueOrTImeNow`?) also adds 1 ms in one the cases.

p.s. also probably better to add 1 nanosecond just in case :rofl: . BUt I still feel this is better of in the place we are using it.